docs/proposals/multi-node/restoration.md

```diff
@@ -0,0 +1,37 @@
+
+If PV is not available, ETCD  BR container connects to the cloud storage and fetches the latest full snapshot and subsequent delta snapshots.
+
+In the particular scenario that we are considering, all the ETCD BR discovers that the respective PVs are unavialble. So, all of them connect to the backup bucket and fetches latest full snapshot and delta snapshots.
+
+All the ETCD BRs form a cluster among themselves. Cluster details are provided by ETCD Druid through configmap. ETCD BR process the configmap and extract the cluster details. Then ETCD BR uses that cluster details to form a temporary cluster to restore the ETCD data directory.
+
+Restoring from the full snapshot only needs the cluster details but no actual ETCD needs to be run.
+
+If there are subsequent delta snapshot after the latest full snapshot, then the delta events needs to be applied from delta snapahot. Applying delta snapshots needs to run an actual ETCD cluster. So an actual ETCD cluster is bootstrapped with cluster details that are already processed earlier. Client ports in the cluster details are changed so that the external clients can't access the cluster while the cluster is restoring from delta snaphots.
+
+The leader in the temporary ETCD cluster applies the delta events. The restoration from delta is co-ordinated by a lease created by ETCD Druid. The leader updates the lease holder identity with code RESTORED once the restoration of the cluster is done. The leading ETCD is stopped then. Rest of the members in the cluster check whether the holder identity of the lease is updated with the code RESTORED or not and if it's updated with the code RESTORED then the respective ETCD is stopped as well.
+
+Main ETCDs are started using restored ETCD Directory in every ETCD BR pod.
+
+## Flow Chart
+
+1. ETCD Druid generates the configmap. This configmap is fed to the ETCD BR pod as ETCD config. Some of the fields in the configmap are supplied as placeholder.
+2. ETCD Druid deploys restoration indicator lease.
+3. ETCD Druid deploys configmap and statefulset with necessary numbers of replicas for ETCD cluster.
+4. ETCD pods are started. ETCD BR checks if there is need of restoration of the cluster.
+5. If there is need of restoration from backup bucket, ETCD configs are processed on each ETCD node and proper cluster details are provided in the ETCD config.
+6. If no delta snapshot is present in the backup bucket, ETCD data directory is prepared from the latest full snapshots in each ETCD pod using cluster details from the ETCD config.
+7. If there are delta snapshots, temporary ETCD is started in each ETCD BR pod and a temporary ETCD cluster is formed using the cluster details from the ETCD config. Temporary ETCD cluster is run on different client port than the client port which is used while running the main ETCD cluster.
```

Comment:
Are these multiple ETCD clusters runnning in parallel to restore the data dir or is this full fledged cluster with all the members joined?